### PR TITLE
Update SR-IOV to `v1.3.0`

### PR DESCRIPTION
**Pull request by mgfritch**
Updates to the latest sriov and nfd chart releases:
* sriov-network-operator `v1.3.0` 
* sriov-nfd `v0.15.6` 

This also modifies the chart version scheme to use the upstream chart's semver instead of the prior scheme (`<version>+up<upstream_version>`). For example, the prior chart version was `1.2.4+up0.1.0` whereas the next published chart version will be `1.3.0`

Open vSwitch networks are currently unsupported. The `OvsNetwork` CRD has been added, but any reference to the `ovs-cni` image has been removed.

### CODE REVIEW

**Comment by mchiappero** [comment]
I think it LGTM, but out of curiosity, for my own understanding: do we need to adds the OvsNetorks CRD anyway?

**Comment by mgfritch** [comment]
> I think it LGTM, but out of curiosity, for my own understanding: do we need to adds the OvsNetorks CRD anyway?

Yes, it's needed because the `sriov-network-operator` will attempt to index the `OVSNetworks` and fail during start-up if the api-resource is not present:

```
unable to create index field for cache  {"error": "no matches for kind \"OVSNetwork\" in version \"sriovnetwork.openshift.io/v1\""}
```

See also: https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/master/main.go#L166-L173

**Comment by mchiappero** [comment]
> Yes, it's needed because the `sriov-network-operator` will attempt to index the `OVSNetwork` and fail during start-up if the api-resource is not present:

Thank you for the explanation! I'll approve once we address the other comment.

File: packages/sriov-network-operator/generated-changes/patch/values.yaml.patch [hunk 1]
```diff
@@ -55,9 +55,6 @@
 +  ibSriovCni: 
 +    repository: rancher/hardened-ib-sriov-cni
 +    tag: v1.1.1-build20240816
-+  ovsCni:
-+    repository: ghcr.io/k8snetworkplumbingwg/ovs-cni-plugin
-+    tag: v0.34.0
```
Comment (hardys) [inline packages/sriov-network-operator/generated-changes/patch/values.yaml.patch]: Is there no downstream build of this image?

File: packages/sriov-network-operator/generated-changes/patch/values.yaml.patch [hunk 2]
```diff
@@ -31,44 +31,43 @@
...
--  operator: ghcr.io/k8snetworkplumbingwg/sriov-network-operator
--  sriovConfigDaemon: ghcr.io/k8snetworkplumbingwg/sriov-network-operator-config-daemon
--  sriovCni: ghcr.io/k8snetworkplumbingwg/sriov-cni
--  ibSriovCni: ghcr.io/k8snetworkplumbingwg/ib-sriov-cni
--  sriovDevicePlugin: ghcr.io/k8snetworkplumbingwg/sriov-network-device-plugin
--  resourcesInjector: ghcr.io/k8snetworkplumbingwg/network-resources-injector
--  webhook: ghcr.io/k8snetworkplumbingwg/sriov-network-operator-webhook
+-  operator: ghcr.io/k8snetworkplumbingwg/sriov-network-operator:v1.3.0
+-  sriovConfigDaemon: ghcr.io/k8snetworkplumbingwg/sriov-network-operator-config-daemon:v1.3.0
+-  sriovCni: ghcr.io/k8snetworkplumbingwg/sriov-cni:v2.8.0
```
[the remark this answers is not in the document]
Comment (hardys) [reply packages/sriov-network-operator/generated-changes/patch/values.yaml.patch:40]: I think the rendering of the patch file is a little confusing - if you look at the [rendered chart ](https://github.com/suse-edge/charts/blob/add925d873264e958f6ab5d15eb8bb5be6c39b1b/charts/sriov-network-operator/1.3.0/values.yaml#L105)you can see it's the downstream 2.8.1 image e.g:

```
  sriovCni:
    repository: rancher/hardened-sriov-cni
    tag: v2.8.1-build20240820
```

**Review by hardys — APPROVED** [review]

File: packages/sriov-network-operator/generated-changes/patch/values.yaml.patch (hunk 2 above)
[the remark this answers is not in the document]
Comment (mchiappero) [reply packages/sriov-network-operator/generated-changes/patch/values.yaml.patch:40]: Yes, sorry I have misinterpreted the diff. I know the actual file was correct, and I could see that, but still I'm not able to understand how this patch was generated, the upstream chart doesn't seem to have versions attached to images...

**Review by mchiappero — APPROVED** [review]

**Review by alknopfler — APPROVED** [review]
LGTM

**Pull request merged**